packages/react-pdf/src/elements/Page.js

```diff
@@ -1,68 +1,63 @@
-import Base from './Base';
-import sizes from '../utils/pageSizes';
+import SubPage from './SubPage';
+
+class Page {
+  parent = null;
+  children = [];
 
-class Page extends Base {
   static defaultProps = {
     size: 'A4',
     orientation: 'portrait',
     style: {},
+    wrap: false,
   };
 
-  getSize() {
-    const { size } = this.props;
-
-    if (typeof size === 'string') {
-      return sizes[size];
-    } else if (Array.isArray(size)) {
-      return size;
-    } else if (typeof size === 'object' && size.width && size.height) {
-      return [size.width, size.height];
-    } else {
-      throw new Error(`Invalid Page size: ${size}`);
-    }
+  constructor(root, props) {
+    this.root = root;
+    this.props = { ...Page.defaultProps, ...props };
+    this.children = [];
+
+    this.addInitialSubpage();
   }
 
-  applyProps(props) {
-    super.applyProps(props);
+  addInitialSubpage() {
+    const newSubpage = new SubPage(this.root, this.props);
```

Comment:
I would put this on the instance instead of using `this.children[0]` everywhere. Makes the shared nature of the children array also a bit more transparent.
`this.initialSubpage = new SubPage(this.root, this.props`